crates/bitwarden-ssh/src/import.rs

```diff
@@ -0,0 +1,214 @@
+use ed25519;
+use pkcs8::{der::Decode, pkcs5, DecodePrivateKey, PrivateKeyInfo, SecretDocument};
+use ssh_key::private::{Ed25519Keypair, RsaKeypair};
+
+use crate::{error::SshKeyImportError, SshKey};
+
+const PKCS1_LABEL: &str = "RSA PRIVATE KEY";
+const PKCS8_UNENCRYPTED_LABEL: &str = "PRIVATE KEY";
+const PKCS8_ENCRYPTED_LABEL: &str = "ENCRYPTED PRIVATE KEY";
+const OPENSSH_LABEL: &str = "OPENSSH PRIVATE KEY";
+
+pub fn import_key(
+    encoded_key: String,
+    password: Option<String>,
+) -> Result<SshKey, SshKeyImportError> {
+    let label = pem_rfc7468::decode_label(encoded_key.as_bytes())
+        .map_err(|_| SshKeyImportError::ParsingError)?;
+
+    match label {
+        PKCS1_LABEL => Err(SshKeyImportError::UnsupportedKeyType),
+        PKCS8_UNENCRYPTED_LABEL => import_pkcs8_key(encoded_key, None),
+        PKCS8_ENCRYPTED_LABEL => import_pkcs8_key(
+            encoded_key,
+            Some(password.ok_or(SshKeyImportError::PasswordRequired)?),
+        ),
+        OPENSSH_LABEL => import_openssh_key(encoded_key, password),
+        _ => Err(SshKeyImportError::ParsingError),
+    }
+}
+
+fn import_pkcs8_key(
```

Comment:
🤔 There's a lot of repetitive `map_err` calls in this method, which leads me to believe that there's a better way to represent this logic. 

For example, extracting the parsing out from the processing of the key would let the parser signal a wide number of error conditions. The importer can then map those parsing failures once as a failed parse.